### PR TITLE
Removed some errors in serializer

### PR DESCRIPTION
**Pull request by manni19682**
Hi, when using serializer I run into some problems and fixed them.

- line 16: 1k buffer is quite small. A response to e.g. 'Get-Printer-Attributes' operation may not fit into so i increased the buffer size to 10k.

- line 38: according to RFC 2910 section 3.3 the 'job-attributes-tag' is part of the Job Template and the Job Object group. I made the mapping an array and added in 'attr' a loop to find the syntax. For example thats necessary for the response to a 'Print-Job' request (RFC 2911 section 3.2.1.2) which needs job-id, job-uri and job-state in a job object attributes group.

- 'resolveAlternates' function: the lookup properties got the name of the callee in 'attributes.js'. Thats 'enumeration' and 'novalue' instead of 'enum' and 'no-value'.

### CODE REVIEW

File: lib/serializer.js [hunk 1]
```diff
@@ -58,7 +58,15 @@ module.exports = function serializer(msg){
...
-		var syntax = attributes[group][name];
+		var syntax;
+
+    if (Array.isArray(group)) {
+			group.forEach((grp) => {
+			  syntax = syntax || attributes[grp][name];
+			});
+		} else {
+			syntax = attributes[group][name];
+		}
```
Comment (williamkapke) [inline lib/serializer.js]: Let's go this route:
```
	var syntax = Array.isArray(group)
		? group.find((grp) => { return attributes[grp][name] })
		: attributes[group][name];
```
Array.find has been available for a long time. As long as the ability to use arrow functions ;) Node will stop supporting the older versions without it soon anyhow.

**Comment by williamkapke** [comment]
Really amazing work here @manni19682 !

Can you change the code as mentioned and also bump the minor version number?
Also, TRY to keep the tab indentation. I really wish it didn't use tabs- but somehow I had tabs turned on when I created these files.

**Comment by manni19682** [comment]
Thank you for the quick answer.

The array find function you prefer returns the array value, not the return value from the callback. So we get e.g. "Job Template" as result instead of a syntax object. This would do the job:

```javascript
  var syntax = Array.isArray(group)
    ? attributes[group.find((grp) => { return attributes[grp][name] })][name]
    : attributes[group][name];
```
But this way we get a quite unspecific exception if the attribute is unknown instead of the "Unknown attribute" exception. A solution with find may be:

```javascript
  var attributeGroup = Array.isArray(group)
    ? attributes[group.find((grp) => { return attributes[grp][name] })]
    : attributes[group];
  if (!attributeGroup) throw "Unknown attribute: " + name;
  
  var syntax = attributeGroup[name];
  if(!syntax) throw "Unknown attribute: " + name;
```

But that's a long piece of code with two times testing for an unknown attribute.

**Comment by williamkapke** [comment]
I think what you changed it to will work fine. Excellent work!

**Pull request merged**